storage/innobase/xtrabackup/src/redo_log.cc

```diff
@@ -610,18 +610,26 @@ uint32_t Archived_Redo_Log_Monitor::get_first_log_block_checksum() const {
   return first_log_block_checksum;
 }
 
-void Archived_Redo_Log_Monitor::skip_for_block(lsn_t lsn, uint32_t no,
-                                               uint32_t checksum) {
+void Archived_Redo_Log_Monitor::skip_for_block(lsn_t lsn, const byte *buf) {
   bool finished = false;
   lsn_t bytes_read = OS_FILE_LOG_BLOCK_SIZE;
 
+  auto no = log_block_get_hdr_no(buf);
+  auto checksum = log_block_get_checksum(buf);
+  auto data_len = log_block_get_data_len(buf);
+
   while (true) {
     auto len = reader.read_logfile(&finished);
     for (auto ptr = reader.get_buffer(); ptr < reader.get_buffer() + len;
          ptr += OS_FILE_LOG_BLOCK_SIZE, bytes_read += OS_FILE_LOG_BLOCK_SIZE) {
       auto arch_block_no = log_block_get_hdr_no(ptr);
       auto arch_block_checksum = log_block_get_checksum(ptr);
-      if (no == arch_block_no && checksum == arch_block_checksum) {
+      auto arch_block_len = log_block_get_data_len(ptr);
+      /* When checksum of redo and archive blocks are different, allow PXB to
+      switch to archive if the data lenght of blocks are different. It
+      indicates they are read during different states.*/
```

Comment:
```suggestion
      /* When checksum of redo and archive blocks are different, allow PXB to
      switch to archive if the data length of blocks is different. This can happen
      when the last block is partially filled in redolog and when it reaches the
      archive file, the same block could be filled more */
```

Reply:
thank you